src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -962,6 +984,19 @@ private void GeneratePropertyGetter<T>(QualifiedType returnType, T decl, Class @
                     ReturnType = decl.QualifiedType
                 };
 
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType == true)
+                {
+                    CSharpTypePrinter _ctp = new CSharpTypePrinter(ctx.Driver);
```

Comment:
We don't prefix locals with '_', that's only in the generated code so that we can avoid naming conflicts. Just 'ctp' is better.